tiledb/sm/storage_manager/context.h

```diff
@@ -111,27 +105,63 @@ class Context {
   /** The class logger. */
   shared_ptr<Logger> logger_;
 
+  /** The storage manager. */
+  StorageManager* storage_manager_;
+
   inline static std::atomic<uint64_t> logger_id_ = 0;
 
   /* ********************************* */
   /*         PRIVATE METHODS           */
   /* ********************************* */
 
   /**
-   * Initializes the thread pools.
+   * Initializes the context with the input config. Called only from
+   * constructor.
+   */
+  Status init(const Config& config);
+
+  /**
+   * Shutdown and delete the storage manager;
+   */
+  void cleanup_storage_manager();
+
+  /**
+   * Get maximum number of threads to use in thread pools, based on config
+   * parameters.
    *
-   * @param config The configuration parameters.
-   * @return Status
+   * @param config The Config to look up max thread information from.
+   * @param max_thread_count (out) Variable to store max thread count.
+   * @return Status of request.
+   */
+  Status get_max_thread_count(const Config& config, uint64_t& max_thread_count);
```

Comment:
This isn't so much a `get_` as a `calc_`.

Reply:
Changed to `compute_max_thread_count()`.